Updated CMakeLists to make it work on Windows

BenPyton opened the pull request:
@hartwork What do you think?
I also copy the necessary dlls to be able to run the exe without having to do it manually each time.
The only thing the windows devs have to do before generating cmake is to add `-DSFML_HOME=path/to/sfml` in command line or from IDE settings.

BenPyton committed:
Updated CMakeLists to make it work on Windows

hartwork reviewed (commented):
Comment deleted, next try later.

hartwork reviewed (commented):
Hi @BenPyton ,

I'm in a better mode for review now, I now understand the idea behind the pull request. I think it's nice move towards improve on the "works out of the box front" for future compiling users.

I think I want to share four small things regarding details:
- Having the command copying all these DLLs in a single line does not seem to be ideal.  If technically possible, I would suggest to make it one filename per line or to to maintain the list of filenames separately in a list variable — something like that. It will be instantly way better to maintain and read.
- The function or need of setting variable `${SFML_DIR}` is not yet clear to me.  Is this a leftover or does this change things behind the scenes in `find_package`?  If it's not a leftover, it would be great to briefly document the intentions and the reason for the specific value.
- Regarding the copying of data, I believe instead the code should have a build parameter where to find its data, likely an absolute path. It may only be a single font file to copy now, but copying data around seems less necessary than DLLs.
- The change kills the trailing newline, as warned by the red Unicode sign.

Best, Sebastian

BenPyton commented:
Hi @hartwork,
Thank you for your feedback.

> I would suggest to make it one filename per line or to to maintain the list of filenames separately in a list variable

Okay, I'll work on it tomorrow.

> The function or need of setting variable `${SFML_DIR}` is not yet clear to me. Is this a leftover or does this change things behind the scenes in `find_package`?

It's not a leftover. It is used by the `find_package` as explained in the [CMake doc](https://cmake.org/cmake/help/v3.0/command/find_package.html):

> Config mode attempts to locate a configuration file provided by the package to be found. A cache entry called <package>_DIR is created to hold the directory containing the file. By default the command searches for a package with the name <package>. If the NAMES option is given the names following it are used instead of <package>. The command searches for a file called <name>Config.cmake or <lower-case-name>-config.cmake for each name specified. 

So, for the package `SFML`, it tries to find the file `SFMLConfig.cmake` at the path given by the variable `SFML_DIR`.

If I don't set it, the configuring error says explicitly to set it.

> copying data around seems less necessary than DLLs.

I understand your point of view.
However, I work in game development, and all game engines I know do have working data folders in the project for development, then copy those folders in the output directory when building the games. Because the data are distributed along the program, and we don't know in advance where it will be installed on the user's computer. (Especially on Windows)
Also the data could be processed for the builds.
It is not a game here, though...

> The change kills the trailing newline, as warned by the red Unicode sign.

What Unicode sign?

Best regards.

hartwork commented:
> > I would suggest to make it one filename per line or to to maintain the list of filenames separately in a list variable
> 
> Okay, I'll work on it tomorrow.

@BenPyton cool!


> So, for the package `SFML`, it tries to find the file `SFMLConfig.cmake` at the path given by the variable `SFML_DIR`.

Thanks for elaborating :+1: 
The current solution seems to depend on the user passing a self-made variable `SFML_HOME` manually, correct?
If so, maybe add a check that it is non-empty and report and fail otherwise.
(I'm also wondering if the existing variable `SFML_ROOT` could do the same in a more standard way.)

> > The change kills the trailing newline, as warned by the red Unicode sign.
> 
> What Unicode sign?

Sorry, I should have been more precise. This guy:
![line_Screenshot_20230211_024308](https://user-images.githubusercontent.com/1577132/218233213-be74da6e-36ed-4b36-8c0e-c45c7d4c4525.png)

BenPyton commented:
Hi @hartwork,
I've updated to add end file's new lines and set a variable for the dlls to copy so we can track them better if changed. 

> maybe add a check that it is non-empty and report and fail otherwise.

Good point. I've updated accordingly.

> (I'm also wondering if the existing variable `SFML_ROOT` could do the same in a more standard way.)

I did some searches and this is what I found about `SFML_ROOT`:
> The CMake system has been changed from SFML 2.5 and we're now using CMake config files and thus SFML_ROOT isn't relevant anymore.

However, I can name it `SFML_ROOT` instead of `SMFL_HOME` anyway.
(I thought of `JAVA_HOME` env variable when I named it)

> Sorry, I should have been more precise. This guy:

Oh, okay, I was on the github phone app and it is not shown there :sweat_smile:...
Indeed I can see it from the web page.

hartwork commented on CMakeLists.txt:
```diff
@@ -1,10 +1,23 @@
+        endif()
+    endif()
+
+    message("Using SFML_HOME=${SFML_HOME}")
```
Just an idea: This call to `message` could use [`STATUS`](https://cmake.org/cmake/help/latest/command/message.html) for a message class.

BenPyton replied:
@hartwork Done.
If you don't see anything else, I can merge the PR. 🙂

hartwork replied:
@BenPyton let's go! :smiley:

hartwork reviewed (commented):
> Hi @hartwork, I've updated to add end file's new lines and set a variable for the dlls to copy so we can track them better if changed.

@BenPyton cool! :+1: 

> > maybe add a check that it is non-empty and report and fail otherwise.
> 
> Good point. I've updated accordingly.

Very nice!

> > (I'm also wondering if the existing variable `SFML_ROOT` could do the same in a more standard way.)
> 
> I did some searches and this is what I found about `SFML_ROOT`:
> 
> > The CMake system has been changed from SFML 2.5 and we're now using CMake config files and thus SFML_ROOT isn't relevant anymore.
> 
> However, I can name it `SFML_ROOT` instead of `SMFL_HOME` anyway. (I thought of `JAVA_HOME` env variable when I named it)

I have 2.5.1 over here and `SFML_ROOT` still has an effect, going into `FIND_SFML_PATHS`:

```console
# qlist media-libs/libsfml | grep cmake/SFML | xargs grep -B2 SFML_ROOT
/usr/lib64/cmake/SFML/SFMLConfig.cmake-set(FIND_SFML_PATHS
/usr/lib64/cmake/SFML/SFMLConfig.cmake-    "${CMAKE_CURRENT_LIST_DIR}/../.."
/usr/lib64/cmake/SFML/SFMLConfig.cmake:    ${SFML_ROOT}
/usr/lib64/cmake/SFML/SFMLConfig.cmake:    $ENV{SFML_ROOT}
```

Let's not use the same name with different semantics please. `SMFL_HOME` is a better choice then. What do you think?
